src/components/ScrollSyncPane.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { Children, ReactElement, cloneElement, useEffect, useRef } from 'react';
+import { useScrollSyncContext } from '../context/ScrollSyncContext';
+
+interface ScrollSyncPaneProps {
+    children: ReactElement;
+    attachTo?: React.RefObject<HTMLElement> | ((node: HTMLElement) => void);
+    group?: string | string[];
+    enabled?: boolean;
+    innerRef?: React.Ref<HTMLElement>;
+}
+
+const ScrollSyncPane: React.FC<ScrollSyncPaneProps> = ({
+    children,
+    attachTo,
+    group = 'default',
+    enabled = true,
+    innerRef,
+}) => {
+    const context = useScrollSyncContext();
+    const childRef = useRef<HTMLElement | null>(null);
+    const nodeRef = useRef<HTMLElement | null>(null);
+
+    const toArray = (groups: string | string[]): string[] => (Array.isArray(groups) ? groups : [groups]);
```

Comment:
Nit but probably a better name would be `castArray`

```suggestion
    const castArray = (groups: string | string[]): string[] => (Array.isArray(groups) ? groups : [groups]);
```

Also this function should not be hoisted in the component since each render will create a new function. Let's move it outside of the component.